test/advanced.js

```diff
@@ -436,6 +436,107 @@ describe('Advanced cases', function(){
         });
       }).error(done);
     });
+    it('hasAndBelongsToMany -- custom pivot tables', function(done) {
+      var Model = thinky.createModel(modelNames[0], {
+        id: String
+      });
+
+      var OtherModel = thinky.createModel(modelNames[1], {
+        id: String
+      });
+
+      var LinkMOM = thinky.createModel(modelNames[0] + '_' + modelNames[1], {
+        id: String,
```

Comment:
You probably need more fields here, I believe we also save the left key and right key in it.
If you set `allowExtra(false)` I believe errors will be throw.